File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
// 1, 2, 3, 4, 5, 6, 7
// 3, 4, 5, 6, 7, 1, 2 

class Solution {
public:
    int findMin(vector<int> &nums) {
        int result = nums[0];
        int left = 0;
        int right = nums.size() - 1;

        while (left <= right) {
            if (nums[left] < nums[right]) {
                result = min(result, nums[left]);
                break;
            }

            int m = (left + right) / 2;
            result = min(result, nums[m]);
            if (nums[m] >= nums[left]) {
                left = m + 1;
            } else {
                right = m - 1;
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        return max(self.maxDepthHelper(1, root.right), self.maxDepthHelper(1, root.left))

    def maxDepthHelper(self, depth: int, root:Optional[TreeNode]):
        if root == None:
            return depth
        
        depth = depth + 1
        return max(self.maxDepthHelper(depth, root.right), self.maxDepthHelper(depth, root.left))
        